src/session.rs

```diff
@@ -119,25 +120,26 @@ impl Session {
     }
 
     /// Perform a digital signature operation
-    fn sign<T: TendermintRequest + Debug>(&mut self, mut request: T) -> Result<Response, Error> {
+    fn sign<R>(&mut self, mut request: R) -> Result<Response, Error>
+    where
+        R: TendermintRequest + Debug,
+    {
         request.validate()?;
 
         let registry = chain::REGISTRY.get();
         let chain = registry.get_chain(&self.config.chain_id).unwrap();
-
-        if let Some(request_state) = &request.consensus_state() {
```

Comment:
It seems like this skips the double signing check if the requested message's consensus state parses as `None`, i.e. if the message doesn't contain a vote:

https://github.com/interchainio/tendermint-rs/blob/566dfb6/src/amino_types/vote.rs#L163

The new logic will return an error in that case, rather than what the old logic did, which is unwrap a `None` value and panic!

https://github.com/interchainio/tendermint-rs/blob/566dfb6/src/amino_types/vote.rs#L144